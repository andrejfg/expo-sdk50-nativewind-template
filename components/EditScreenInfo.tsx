import React from 'react';
import { Text, View } from 'react-native';
import { ExternalLink } from './ExternalLink';
import { MonoText } from './StyledText';

export default function EditScreenInfo({ path }: { path: string }) {
  return (
    <View>
      <View className='items-center mx-10'>
        <Text
          className='text-base leading-6 text-cente text-[rgba(0,0,0,0.8)] dark:text-[rgba(255,255,255,0.8)]'
>
          Open up the code for this screen:
        </Text>

        <View
          className='rounded-md px-4 my-2 bg-[rgba(0,0,0,0.05)] dark:bg-[rgba(255,255,255,0.05)]'>
          <MonoText>{path}</MonoText>
        </View>

        <Text
          className='text-base leading-6 text-center text-[rgba(0,0,0,0.8)] dark:text-[rgba(255,255,255,0.8)]'
>
          Change any of the text, save the file, and your app will automatically update.
        </Text>
      </View>

      <View className='mt-5 mx-4 flex items-center'>
        <ExternalLink
          className='py-4'
          href="https://docs.expo.io/get-started/create-a-new-app/#opening-the-app-on-your-phonetablet">
          <Text className='text-center text-link'>
            Tap here if your app doesn't automatically update after making changes
          </Text>
        </ExternalLink>
      </View>
    </View>
  );
}
