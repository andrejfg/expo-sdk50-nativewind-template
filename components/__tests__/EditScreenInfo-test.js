import * as React from 'react'
import renderer from 'react-test-renderer'

import EditScreenInfo from '../EditScreenInfo'

it(`renders correctly`, () => {
  const tree = renderer.create(<EditScreenInfo path="app/modal.tsx" />).toJSON()

  expect(tree).toMatchSnapshot()
})
