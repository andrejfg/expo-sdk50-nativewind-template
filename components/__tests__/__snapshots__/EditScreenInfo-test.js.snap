// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={[]}
  >
    <Text
      onLayout={[Function]}
      style={[]}
    >
      Open up the code for this screen:
    </Text>
    <View
      onLayout={[Function]}
      style={[]}
    >
      <Text
        style={
          [
            undefined,
            {
              "fontFamily": "SpaceMono",
            },
          ]
        }
      >
        app/modal.tsx
      </Text>
    </View>
    <Text
      onLayout={[Function]}
      style={[]}
    >
      Change any of the text, save the file, and your app will automatically update.
    </Text>
  </View>
  <View
    onLayout={[Function]}
    style={[]}
  >
    <Text
      href="https://docs.expo.io/get-started/create-a-new-app/#opening-the-app-on-your-phonetablet"
      onLayout={[Function]}
      onPress={[Function]}
      role="link"
      style={[]}
    >
      <Text
        onLayout={[Function]}
        style={[]}
      >
        Tap here if your app doesn't automatically update after making changes
      </Text>
    </Text>
  </View>
</View>
`;
